frontend/src/firebase/firebaseInteractor.ts

```diff
@@ -134,6 +134,39 @@ export default class FirebaseInteractor {
     });
   }
 
+  /**
+   * Creates a new assessment with all of the incorrect words from a previous assessment.
+   */
+  async createAssessmentFromPreviousAssessment(setId: string) {
+    // hardcode a single assessment for now
+    // later we can infer the assessment id from the setId
+    let assessmentId = "oiBN8aE5tqEFK2gXJUpl";
+    let results = await this.db
+      .collection("assessments")
+      .doc(assessmentId)
+      .collection("results")
+      .get();
+    if (results == null) {
+      throw new Error(`Assessment with id ${assessmentId} does not exist`);
+    }
+
+    let wordList: string[] = [];
+    results.forEach(function (result) {
+      if (!result.get("correct")) {
```

Comment:
It generally makes more sense to call `data` on each result, which gives a plain javascript object, that you can call `.correct` on. This works as  well tho

Reply:
sure, updated!